fvtest/compilertriltest/SimplifierFoldAbsNegTest.cpp

```diff
@@ -0,0 +1,237 @@
+                                                  "   (%sabs                      "
+                                                  "    (%sabs (%sload parm=0))))))",
+                  nameForType<TypeParam>(), nameForType<TypeParam>(),
+                  prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>());
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+    NoAbsAbsIlVerifier verifier;
+
+    ASSERT_EQ(0, compiler.compileWithVerifier(&verifier)) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<TypeParam(*)(TypeParam)>();
+
+    // Invoke the compiled method, and assert the output is correct.
+    for (auto test : dataForType<TypeParam>()) {
+        EXPECT_EQ(std::abs(test), entry_point(test));
+    }
+}
+
+
+class NoAbsNegIlVerifier : public SimplifierFoldAbsNegTestIlVerifierBase
+   {
+   protected:
+   virtual int32_t verifyNode(TR::Node *node) override
+      {
+      if (node->getOpCode().isAbs())
+         {
+         if (node->getChild(0)->getOpCode().isNeg())
+            {
+            return 1;
+            }
+         }
+      return 0;
+      }
+   };
+
+/*
+ * method(T parameter)
+ *   return abs(neg(parameter));
+ */
+TYPED_TEST(SimplifierFoldAbsNegTest, FoldAbsNeg) {
+    char inputTrees[256];
+    std::snprintf(inputTrees, sizeof(inputTrees), "(method return=%s args=[%s]    "
+                                                  " (block                        "
+                                                  "  (%sreturn                    "
+                                                  "   (%sabs                      "
+                                                  "    (%sneg (%sload parm=0))))))",
+                  nameForType<TypeParam>(), nameForType<TypeParam>(),
+                  prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>(), prefixForType<TypeParam>());
+
+    auto trees = parseString(inputTrees);
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+    NoAbsNegIlVerifier verifier;
+
+    ASSERT_EQ(0, compiler.compileWithVerifier(&verifier)) << "Compilation failed unexpectedly\n" << "Input trees: " << inputTrees;
+
+    auto entry_point = compiler.getEntryPoint<TypeParam(*)(TypeParam)>();
+
+    // Invoke the compiled method, and assert the output is correct.
+    for (auto test : dataForType<TypeParam>()) {
+        EXPECT_EQ(std::abs(test), entry_point(test));
+    }
+}
+
+
+class NoNegNegIlVerifier : public SimplifierFoldAbsNegTestIlVerifierBase
+   {
+   protected:
+   virtual int32_t verifyNode(TR::Node *node) override
+      {
+      if (node->getOpCode().isNeg())
+         {
+         if (node->getChild(0)->getOpCode().isNeg())
+            {
+            return 1;
+            }
+         }
+      return 0;
```

Comment:
Couldn't this be simplified to `return node->getOpCode().isNeg() && node->getChild(0)->getOpCode().isNeg()`?

Reply:
Good idea, updated.